Set auto pause play

amandapi opened the pull request.

amandapi committed:
A Timed Version

amandapi committed:
Set Auto Pause and Play

nerdyverde commented on Swiftris/Swiftris/AppDelegate.swift:
```diff
@@ -16,31 +16,40 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
-    }
+        NSLog("From AppDelegate: did resign active")
+
+        NSNotificationCenter.defaultCenter().postNotificationName("swiftrisDidEnterBackground", object: self)
```
Posting a custom notification from `applicationWillResignActive()` is not necessary. The app will post a `UIApplicationWillResignActiveNotification` notification after the method finishes. It is this notification that you are listening for in GameViewController.

nerdyverde commented on Swiftris/Swiftris/AppDelegate.swift:
```diff
@@ -16,31 +16,40 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Restart any tasks that were paused (or not yet started) while the application was inactive. If the application was previously in the background, optionally refresh the user interface.
+        NSLog("From AppDelegate: did become active")
+        
+        NSNotificationCenter.defaultCenter().postNotificationName("swiftrisDidEnterForeground", object: self)
```
Same thing as above. Posting a custom notification is not necessary in this scenario.

nerdyverde commented on Swiftris/Swiftris/GameViewController.swift:
```diff
@@ -43,11 +51,15 @@ class GameViewController: UIViewController, SwiftrisDelegate, UIGestureRecognize
+    let path = NSBundle.mainBundle().pathForResource("theme", ofType:"mp3")
+    let fileURL = NSURL(fileURLWithPath: path!)
+    player = AVAudioPlayer(contentsOfURL: fileURL, error: nil)
+    player.numberOfLoops = -1  // loop theme infinitely
```
Good to see that we're looping the theme music indefinitely now :)

nerdyverde commented on Swiftris/Swiftris/GameViewController.swift:
```diff
@@ -17,14 +17,22 @@ class GameViewController: UIViewController, SwiftrisDelegate, UIGestureRecognize
     var player:AVAudioPlayer!
     var isPause: Bool = false
+    var timer = NSTimer()     // for countdown timer
+    var timerCount = 90       // for countdown timer
```
This is probably something that should be user-configurable. Some users may want to play shorter games while other may want to play longer games. Let's discuss how you might go about providing this option for your users during our next call.

nerdyverde commented:
Overall the changes look good Amanda! I made some specific suggestions as inline comments. In addition, I would like to see the user provided with the option of choosing either the classic version of the game or the timed version. This will require creating some kind of start screen for the game where the user can choose the appropriate options.

nerdyverde commented on Swiftris/Swiftris/GameViewController.swift:
```diff
@@ -104,6 +118,29 @@ class GameViewController: UIViewController, SwiftrisDelegate, UIGestureRecognize
+        }
+    }
+    
+    func swiftrisDidEnterForeground() {  // read with NSNotification
```
You may want to reconsider un-pausing the game automatically when the application becomes active again. This may take the user by surprise and end up costing them the game. Instead, I think it would be better to allow the user to un-pause the game when they are ready.

nerdyverde committed:
Update to auto-pause

Changed behaviour so that button state updates when game is
automatically paused after receiving
UIApplicationWillResignActiveNotification. Removed method for handling
un-pause automatically.

nerdyverde commented:
As we discussed during our call, feel free to merge this PR and then create a new branch to continue work on the following:
1. Blanking out the tetrominos when the game is paused.
2. Adding a SetupViewController to allow the user to select either Classic or Timed game modes.

amandapi commented:
OK!
